refbox/src/tournament_manager/mod.rs

```diff
@@ -1794,11 +2022,20 @@ impl TournamentManager {
         }
     }
 
+    pub fn in_score_confirm_pause(&self) -> bool {
+        self.time_pause_confirmation.is_some()
+    }
+
     pub fn generate_snapshot(&mut self, now: Instant) -> Option<GameSnapshot> {
         trace!("Generating snapshot");
         let cur_time = self.game_clock_time(now)?;
         trace!("Got current time: {cur_time:?}");
-        let secs_in_period = cur_time.as_secs().try_into().ok()?;
+        let zero_for_pause = Duration::ZERO;
+        let secs_in_period = if self.time_pause_confirmation.is_some() {
+            zero_for_pause.as_secs().try_into().ok()?
+        } else {
+            cur_time.as_secs().try_into().ok()?
+        };
```

Comment:
I believe that with the switch to stopping the clock, this change can be removed. That way, we will also correctly show the current time in sudden death if the pause happens in sudden death (rather than showing sudden death 0:00)